docs/features/pointing_device.md

```diff
@@ -616,6 +643,249 @@ layer_state_t layer_state_set_user(layer_state_t state) {
 
 ```
 
+### High Resolution Drag Scroll
+
+This basic implementation of high resolution drag-scroll sends wheel data out at a throttled rate to avoid overloading the host computer's input buffer. Add `#define WHEEL_EXTENDED_REPORT` to `config.h` in order to avoid overflow errors.
+
+```c
+// Modify this value to adjust the rate at which mouse reports are sent while drag scrolling
+#define SCROLL_RATE 16
+
+enum custom_keycodes {
+    DRAG_SCROLL = SAFE_RANGE,
+};
+
+bool set_scrolling = false;
+uint32_t last_scroll_time = 0;      // last time a mouse report was sent
+int16_t scroll_accumulated_h = 0;   // accumulated scroll values
+int16_t scroll_accumulated_v = 0;   // accumulated scroll values
+
+// Function to handle mouse reports and perform drag scrolling
+report_mouse_t pointing_device_task_user(report_mouse_t mouse_report) {
+    
+    // Check if drag scrolling is active
+    if (!set_scrolling) { return mouse_report; }
+    
+    // Accumulate scroll values based on mouse movement
+    scroll_accumulated_h += mouse_report.x;
+    scroll_accumulated_v += mouse_report.y;
+    
+    // Clear the X and Y values of the mouse report
+    mouse_report.x = 0;
+    mouse_report.y = 0;
+    
+    // Check to see if it's time to send another mouse report; if not, stop here
+    if (timer_elapsed32(last_scroll_time) < SCROLL_RATE) {
+        mouse_report.h = 0;
+        mouse_report.v = 0;
+        return mouse_report;
+    }
+    
+    // Keep track of when we last sent a mouse report
+    last_scroll_time = timer_read32();
+    
+    // Assign accumulated scroll values to the mouse report
+    mouse_report->h = scroll_accumulated_h;
+    mouse_report->v = scroll_accumulated_v;
+    
+    // Reset accumulated scroll values
+    scroll_accumulated_h = 0;
+    scroll_accumulated_v = 0;
+    
+    // Return modified mouse report
+    return mouse_report;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (keycode == DRAG_SCROLL && record->event.pressed) {
+        set_scrolling = !set_scrolling;
+    }
+    return true;
+}
+```
+
+### Advanced High Resolution Drag Scroll
+
+This implementation of high resolution drag-scroll implements several features: throttling, rescaling, smoothing, and axis-snapping. As in the previous example, make sure to add `#define WHEEL_EXTENDED_REPORT` to `config.h` in order to avoid overflow errors.
```

Comment:
This is a very large example section and is much more than "show how to interface with QMK." I can definitely see the value of an existing fully-featured implementation for something as complex as comfortable scrolling with a ball, but the docs are not the place for it. Best-case scenario this gets copied however many times and some are out of date, have extra features added, or break.

IMO this should go in something similar to a `process_record` function but for `pointing_device_task`. We currently have `pointing_device_task_kb` -> `pointing_device_task_user`, while `process_record` calls the similar `process_record_kb` -> `process_record_user` - it shouldn't be too messy to mirror [`process_record`](https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.c#L266) for [pointing devices.](https://github.com/qmk/qmk_firmware/blob/master/quantum/pointing_device/pointing_device.c#L288) Then this feature can be expanded and options added and such as well when people need something new for themselves.

Reply:
I don't think I 100% understand what you're suggesting, but I'll try my best:

I see that there's a function `pointing_device_task` defined in `pointing_device.c`. This function calls several other functions (e.g. `pointing_device_adjust_by_defines` and `pointing_device_task_kb`) to sequentially process the mouse report prior to handing it off to be sent to the host computer. I think it would be sensible to add a function `pointing_device_drag_scroll` to this "pipeline", which would provide a default implementation of drag scroll. 

Ideally, this function would automatically adjust its behavior based on whether high resolution scrolling is enabled. You could also use compiler flags to control aspects of behavior such as throttling, rescaling, smoothing, axis-snapping, etc. There would also need to be a set of functions `drag_scroll_on`, `drag_scroll_off`, `drag_scroll_toggle`, and `is_drag_scroll_on`.

Does this sound consistent with what you're proposing?

Reply:
Yes, exactly - just adding it to the baby `pointing_device_task` chain in the same way we do for the massive `process_record_quantum` keyboard one. We don't have the former formalized as being the same design pattern AFAIK, but it looks like that was the intent to me.

Moving that to a separate PR as you are doing is probably best. The 60Hz limitation can be a separate one (it is not necessary on Linux and really a driver/application issue. . . mice can report cursor movement at 10kHz) which makes doc examples not necessary at all. High-res scrolling itself doesn't need anything special.